### DISCUSSION TITLE
Typescript withZod

### DISCUSSION THREAD

**Discussion by barbinbrad** [airjp73/rvf#259] [General]
Thought this might be useful to someone else. It's a utility type for extracting the type of your validator. I use this to define my props that get passed to my `<ValidatedForm>`:

```ts
export type TypeOfValidator<U extends Validator<any>> = U extends Validator<
  infer T
>
  ? T
  : unknown;
```

Here's how to use it:

```tsx
import { bigValidator } from '~/validators'
import type { TypeOfValidator } from '~/types'

type FormProps = {
  defaultValues: TypeOfValidator<typeof bigValidator>
}
```

**Comment by airjp73** [comment]
To save you a little bit of code, this is pretty much exactly what the `ValidatorData` type exported from `remix-validated-form` does.

**Reply by barbinbrad** [reply]
Thanks so much! I was looking around for something like this, and couldn't find it. Really nice work on the library, btw. This is by far the nicest approach to forms that I've ever used.
